File: CSharp/OOP/MixTapeFramework/Pages/LoginPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium;

namespace MixTapeFramework.Pages
{
    public class LoginPage : PageBase
    {
        public override string Url
        {
            get
            {
                return base_url + "/#/login";
            }
        }

        public LoginPage(IWebDriver driver) : base(driver)
        {
        }

        #region Elements
        public IWebElement username
        {
            get
            {
                return driver.FindElement(By.Id("inputEmail3"));
            }
        }

        public IWebElement password
        {
            get
            {
                return driver.FindElement(By.Id("inputPassword3"));
            }
        }

        public IWebElement loginButton
        {
            get
            {
                return driver.FindElement(By.CssSelector("form > button"));
            }
        }

        public IWebElement rememberMe
        {
            get
            {
                return driver.FindElement(By.CssSelector("input[type='checkbox'"));
            }
        }

        public IWebElement createAccount
        {
            get
            {
                return driver.FindElement(By.LinkText("Create an account*"));
            }
        }
        #endregion

        public LoginCommand Login(string username)
        {
            return new LoginCommand(username, this);
        }

        public void LoginWithDefault()
        {
            username.SendKeys("testuser");
            password.SendKeys("password");
            loginButton.Click();
        }
    }

    public class LoginCommand
    {
        private readonly string username;
        private readonly LoginPage loginPage;
        private string password;

        public LoginCommand(string username, LoginPage loginPage)
        {
            this.username = username;
            this.loginPage = loginPage;
        }

        public LoginCommand WithPassword(string password)
        {
            this.password = password;
            return this;
        }

        public void Login()
        {
            loginPage.username.SendKeys(username);
            loginPage.password.SendKeys(password);
            loginPage.loginButton.Click();
        }
    }
}

File: Java/src/main/PageObjectsAndTools/LoginPage.java
package PageObjectsAndTools;
<<<<<<< Updated upstream
=======

import PageObjectsAndTools.BaseObject;
>>>>>>> Stashed changes
import PageObjectsAndTools.PageObject;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Action;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;

import javax.script.ScriptContext;


public class LoginPage extends PageObject {

    public LoginPage(WebDriver driver, WebDriverWait wait) {
        super(driver, wait);
    }

    public WebElement getNavItem(){return driver.findElement(By.cssSelector("[ngbdropdown]"));}

    public WebElement getButton() {
        return driver.findElement(By.cssSelector("form > button"));
    }

    public WebElement getDropDownItem() {return driver.findElement(By.cssSelector(".dropdown-item"));}

    public boolean buttonBlock() {return getButton().getAttribute("disabled").isEmpty();}

    public boolean dropDownShow() {return !getButton().getAttribute("class").contains("show");}

    public void actionNavItem() {
        Actions builder = new Actions(driver);
        Action action = builder
                .click(getNavItem())
                .click(getDropDownItem())
                .build();
        action.perform();
    }

    /**
     * takes you to the login page url
     */

    public void goToMixTapeLogin() {
        driver.navigate().to(baseURL + "#/login");
    }

    /**
     * @return username field
     */
    public WebElement UserNameLogin(){
        WebElement username = driver.findElement(By.id("inputEmail3"));
        return username;
    }

    /**
     * @return the password field
     */

    public WebElement PasswordLogin(){
        WebElement password = driver.findElement(By.id("inputPassword3"));
        return password;
    }

    /**
     * @return the "sign in" button
     */

    public WebElement SignInButton(){
        WebElement button = driver.findElement(By.cssSelector("form > button"));
        return button;
    }

    /**
     * creates the login for Madilena's mixtape page
     */
    public void MadilenaRobotLogin(){
        setLogin("MadilenaM" , "NickSurfsBirdRock");
    }

    /**
     * sends login credentials to the username and password fields
     * @param UsernameSearchText the username you want to input
     * @param PasswordSearchText the password you want to input
     */
    public void setLogin(String UsernameSearchText , String PasswordSearchText) {
        goToMixTapeLogin();
        UserNameLogin().clear();
        UserNameLogin().sendKeys(UsernameSearchText);
        PasswordLogin().clear();
        PasswordLogin().sendKeys(PasswordSearchText);
        SignInButton().click();

        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            System.out.println("DID NOT WAIT BEFORE CLICKING ENTER");
        }
    }
}
